janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphBlueprintsGraph.java

```diff
@@ -97,6 +97,7 @@ public JanusGraphTransaction getCurrentThreadTx() {
 
     @Override
     public synchronized void close() {
+        txs.remove();
```

Comment:
@mad @porunov 
This only closes the ThreadLocal of the current thread. Doesn't this still cause ThreadLocal leak, when the user neither calls `graph.tx().close()` explicitly nor wraps `graph.tx()` in a try-with-resources clause?